File: Day12/Car.test.js
const Car = require("./Car") 

test("Test that can has a status", () =>{
    const car = new Car();
    expect(car.getFuelLevel()).toBe(0)


})

test("Test that a car can start", () => {
    const car = new Car();
    car.refuelCar(20);
    car.startCar();
    expect(car.checkIsRunning()).toBe(true)
})

test("Test that a car can stop", ()=> {
    const car = new Car();
    car.stopCar();
    expect(car.checkIsRunning()).toBe(false)
})

test("Test that a car can be refueled", ()=> {
    const car = new Car();
    car.refuelCar(20)
    expect(car.getFuelLevel()).toBe(20);
})

test("Test that a car can drive", ()=>{
    const car = new Car();
    car.refuelCar(20)
    car.startCar()
    car.driveCar(20)
    expect (car.getFuelLevel()).toBe(18.3333333333333332)
})

File: Day12/CarTest.java
import org.junit.jupiter.api.Test;
import static org.junit.jupiter.api.Assertions.*;

public class CarTest{

	@Test
	public void testACarStatus(){
		Car car = new Car();
		boolean actual = car.getCarStatus();		
		assertFalse( actual);
	}


	@Test
	public void testThatACarCanStart(){
	Car car = new Car();
	car.refuelCar(20);
	car.startCar();
	boolean actual = car.getCarStatus();		
	assertTrue(actual);

	}


	@Test
	public void testThatCarCanStop(){
	Car car = new Car();
	car.stopCar();
	boolean actual = car.getCarStatus();
	assertFalse(actual);



	}

	@Test
	public void testThatCarCanBeRefueled(){
	Car car = new Car();
	car.refuelCar(20);
	double actual = car.getFuelLevel();
	double expected = 20.0;
	assertEquals(actual,expected);

	}

	@Test
	public void testThatACarCanDrive(){
	Car car = new Car();
	car.refuelCar(20);
	car.startCar();
	car.drive(20);
	double actual = car.getFuelLevel();
	double expected = 18.333333333333332;
	assertEquals(actual,expected);

		
	}


	




}